File: articles/sensitivity-analysis-and-scenario-comparison/python/run_all_sensitivity_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Sensitivity Analysis and Scenario Comparison."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "one_way_sensitivity.py",
    "multi_way_sensitivity.py",
    "threshold_analysis.py",
    "scenario_robustness_profiles.py",
    "regret_analysis.py",
    "probabilistic_sensitivity.py",
    "key_driver_diagnostics.py",
    "decision_record_exporter.py",
    "sensitivity_analysis_scenario_comparison_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python sensitivity-analysis workflows completed.")


if __name__ == "__main__":
    main()

File: articles/sensitivity-analysis-and-scenario-comparison/r/run_all_sensitivity_workflows.R
# run_all_sensitivity_workflows.R
# Run all R workflows for Sensitivity Analysis and Scenario Comparison.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "sensitivity_analysis_scenario_comparison_workflow.R",
  "ranking_stability_profiles.R",
  "scenario_robustness_report.R",
  "threshold_analysis_tables.R",
  "regret_summary_tables.R",
  "probabilistic_sensitivity_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R sensitivity-analysis workflows completed.")
